tests/rptest/redpanda_cloud_tests/omb_validation_test.py

```diff
@@ -431,10 +440,94 @@ def test_common_workload(self):
+            benchmark.wait(timeout_sec=benchmark_time_min * 60)
+
+            #benchmark.check_succeed()
+            # check if omb gave errors, but don't process metrics
+            #benchmark.check_succeed(validate_metrics=True, )
+            is_valid, validation_results = benchmark.check_succeed(
+                raise_exceptions=False)
+
+            if is_valid:
+                self.logger.info(
+                    "\n\nBenchmark passed without significant latency issues.\n"
+                )
+                for result in validation_results:
+                    self.logger.info(f"Result is: {result}")
+                break
+
+            else:
+                self.logger.info(
+                    f"\nBenchmark test attempt {try_count} failed.\n")
+                for result in validation_results:
+                    self.logger.info(f"Result is: {result}")
+                self.logger.info(
+                    "Checking test results for possible latency spikes.")
+
+                #latency_metrics = benchmark.check_succeed(validate_metrics=False, return_latency_metrics=True, raise_exceptions=False)
+
+                latency_metrics = {
+                    key: benchmark.metrics[key]
+                    for key in self.LATENCY_SERIES_AND_MAX
+                    if key in benchmark.metrics
+                }
+
+                if latency_metrics:  # Ensure latency_metrics is not None or empty
+                    self.logger.info("\nLatency metrics for spikes detection:")
+                    self.logger.info(
+                        "\n========================================")
+                    for key, value in latency_metrics.items():
+                        # Assuming value is a list of latency values for each series
+                        series_values = ", ".join(str(v) for v in value)
+                        self.logger.info(f"    {key}: [{series_values}]")
+                    self.logger.info(
+                        "\n========================================")
+                else:
+                    self.logger.info("No latency metrics returned.")
+
+                # Prepare the expected_max_latencies dictionary for spike detection
+                expected_max_latencies_for_detection = {
+                    series_key:
+                    OMBValidationTest.EXPECTED_MAX_LATENCIES[config_key]
+                    for series_key, config_key in
+                    OMBValidationTest.LATENCY_SERIES_AND_MAX.items()
+                }
+
+                spikes_detected = benchmark.detect_spikes_by_percentile(
+                    latency_metrics,
+                    expected_max_latencies=expected_max_latencies_for_detection
+                )
+
+                if spikes_detected and try_count < max_retries:
+                    self.logger.info(
+                        f"\n\nPossible latency spikes detected.\n\nPreparing for attempt {try_count+1}/{max_retries}."
```

Comment:
include the number of spikes detected in the log line, perhaps any ohter useful info about the spikes? or is that already logged elsewhere?

Reply:
Yes this was already logged from the function and I added more logs:

Example of one of the cases:

Checking for spikes in series: endToEndLatency50pct
Latency data for endToEndLatency50pct series is within allowed max values
Checking for spikes in series: endToEndLatency75pct
Latency data for endToEndLatency75pct series is within allowed max values
Checking for spikes in series: endToEndLatency99pct
Isolated spike detected at index 14. Value: 1062.759
Spikes detected in series: endToEndLatency99pct.
Latency data for endToEndLatency99pct series is within allowed max values
Checking for spikes in series: endToEndLatency999pct
Sustained high latency detected in series endToEndLatency999pct, not considered a spike. Sequence starts at index 1 and ends at index 2. Values: [890.223, 132.848]
Isolated spike detected at index 14. Value: 1280.239
Ending isolated spike detected from index 29 to 29. Values: [209.906]
Spikes detected in series: endToEndLatency999pct.
Sustained high latency detected in series: endToEndLatency999pct. Advising against retry.

No retry advised due to sustained high latency in one or more series.